src/modules/previewpane/powerpreview/settings.cpp

```diff
@@ -60,12 +68,107 @@ namespace PowerPreviewSettings
 		}
 	}
 
+	LONG FileExplorerPreviewSettings::SetRegistryValue() const
+	{
+		HKEY hKey = HKEY_CURRENT_USER;
+		const REGSAM WRITE_PERMISSION = KEY_WRITE;
+		DWORD options = 0;
+		HKEY OpenResult;
+
+		LONG err = RegOpenKeyEx(hKey, this->GetSubKey(), options, WRITE_PERMISSION, &OpenResult); 
+
+		if (err == ERROR_SUCCESS)
+		{
+			err = RegSetValueExW(
+					OpenResult,
+					this->GetCLSID(),
+					0,
+					REG_SZ,
+					(LPBYTE)this->GetDisplayName().c_str(),
+					this->GetDisplayName().length() * sizeof(TCHAR));
+
+			if (err != ERROR_SUCCESS)
+			{
+				return err;
+			}
+		}
+		else
+		{
+			return err;
+		}
+        	RegCloseKey(OpenResult);
+	}
+
+	LONG FileExplorerPreviewSettings::RemvRegistryValue() const
+	{
+		HKEY hKey = HKEY_CURRENT_USER;
+		const REGSAM WRITE_PERMISSION = KEY_WRITE;
+		DWORD options = 0;
+		HKEY OpenResult;
+
+		LONG err = RegOpenKeyEx(hKey, this->GetSubKey(), options, WRITE_PERMISSION, &OpenResult); 
+		if (err == ERROR_SUCCESS)
+		{
+			err = RegDeleteKeyValueW(
+				OpenResult,
+				NULL,
+				this->GetCLSID());
+
+			if (err != ERROR_SUCCESS)
+			{
+				return err;
+			}
+		}
+		else
+		{
+			return err;
+		}
+        	RegCloseKey(OpenResult);
+	}
+
+	bool FileExplorerPreviewSettings::GetRegistryValue() const
+	{
+		HKEY OpenResult;
+		LONG err = RegOpenKeyEx(
+			HKEY_CURRENT_USER,
+			this->GetSubKey(), 
+			0, 
+			KEY_READ,
+			&OpenResult);
+
+		if (err != ERROR_SUCCESS)
+		{
+			return false;
+		}
+		else
+		{
+			DWORD dataType;
+			WCHAR value[255];
+			PVOID pvData = value;
+			DWORD size = sizeof(value);
+
+			err = RegGetValueW(
+					OpenResult, 
+					NULL,
+					this->GetCLSID(),
+					RRF_RT_ANY,
+					&dataType,
+					pvData,
```

Comment:
Where is this being used? If it is not being used anywhere then [we could just set it to NULL](https://docs.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-reggetvaluew). However, if that is being used, how do we know that the data associated with the registry value is only 255 wide chars? I found [limits](https://docs.microsoft.com/en-us/windows/win32/sysinfo/registry-element-size-limits) for various registry elements but not for the data associated with value. Could you clarify the same?

Reply:
Not related to the PR, since the value is not being used, but I was also wondering the same thing.
What people do to set the size of the buffer is to make two consecutive calls. In the first call you pass a NULL buffer and you get the size of the buffer then allocate the buffer with the correct size and you make the second call and pass the correctly allocated buffer